apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java

```diff
@@ -460,38 +418,47 @@ public String toString() {
   }
 
   /**
-   * Per <code>/dev/design-decisions-readme.md</code>, this application accepts its configuration
-   * via environment variables. Read those in, and build an {@link AppConfiguration} instance from
-   * them.
+   * Build a {@link ConfigLoader} that accounts for all possible sources of configuration
+   * information. The provided function is used to look up environment variables so that these can
+   * be simulated in tests without having to fork a process.
+   *
+   * <p>{@see LayeredConfiguration#createConfigLoader} for possible sources of configuration
+   * variables.
+   *
+   * @param getenv function used to access environment variables (provided explicitly for testing)
+   * @return appropriately configured {@link ConfigLoader}
+   */
+  static ConfigLoader createConfigLoader(Function<String, String> getenv) {
+    return LayeredConfiguration.createConfigLoader(DEFAULT_CONFIG_VALUES, getenv);
+  }
+
+  /**
+   * Load configuration variables using the provided {@link ConfigLoader} instance and build an
+   * {@link AppConfiguration} instance from them.
    *
    * <p>As a convenience, this method will also verify that AWS credentials were provided, such that
    * {@link DefaultCredentialsProvider} can load them. If not, an {@link AppConfigurationException}
    * will be thrown.
    *
-   * @return the {@link AppConfiguration} instance represented by the configuration provided to this
-   *     application via the environment variables
-   * @throws AppConfigurationException An {@link AppConfigurationException} will be thrown if the
-   *     configuration passed to the application are incomplete or incorrect.
-   */
-  static AppConfiguration readConfigFromEnvironmentVariables() {
-    int hicnHashIterations = readEnvIntPositiveRequired(ENV_VAR_KEY_HICN_HASH_ITERATIONS);
-    byte[] hicnHashPepper = readEnvBytesRequired(ENV_VAR_KEY_HICN_HASH_PEPPER);
-    int hicnHashCacheSize =
-        readEnvIntOptional(ENV_VAR_KEY_HICN_HASH_CACHE_SIZE).orElse(DEFAULT_HICN_HASH_CACHE_SIZE);
-
-    int loaderThreads = readEnvIntPositiveRequired(ENV_VAR_KEY_LOADER_THREADS);
-    boolean idempotencyRequired = readEnvBooleanRequired(ENV_VAR_KEY_IDEMPOTENCY_REQUIRED);
+   * @param config used to load configuration values
+   * @return the {@link AppConfiguration} instance
+   * @throws ConfigException will be thrown if the configuration passed to the application are
+   *     incomplete or incorrect.
+   */
+  static AppConfiguration loadConfig(ConfigLoader config) {
+    int hicnHashIterations = config.positiveIntValue(ENV_VAR_KEY_HICN_HASH_ITERATIONS);
+    byte[] hicnHashPepper = config.hexBytes(ENV_VAR_KEY_HICN_HASH_PEPPER);
+    int hicnHashCacheSize = config.intValue(ENV_VAR_KEY_HICN_HASH_CACHE_SIZE);
+
```

Comment:
The `Optional.orElse` pattern has been replaced by simple defaults provided to the `ConfigLoader` when it is created.  Code here can just assume that such config variables will always have a value defined.